### PR TITLE
Connects Falkonry/tracker#814 Java Facts API Changes

### PR DESCRIPTION
**Pull request by dilpeshjain**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -847,13 +847,21 @@ Usage:
...
+File Sample
+{"time" : "2011-03-26T12:00:00.000Z", "thing" : "thing1", "end" : "2012-06-01T00:00:00.000Z", "Health" : "Normal"}
+{"time" : "2014-02-10T23:00:00.000Z", "thing" : "thing1", "end" : "2014-03-20T12:00:00.000Z", "Health" : "Spalling"}
 ```java
     import com.falkonry.client.Falkonry;
 
     Falkonry falkonry = new Falkonry("http://localhost:8080", "auth-token");
-
-    String data = "{\"time\" : \"2011-03-26T12:00:00Z\", \"entities\" : \"entity1\", \"end\" : \"2012-06-01T00:00:00Z\", \"Health\" : \"Normal\"}";
+	Map<String, String> options = new HashMap<String, String>();
```
Comment (aniket-amrutkar) [inline README.md]: • Add one test case for using the tagIdentifier
• Add one test case for sending the additional tag while uploading facts

File: src/main/java/com/falkonry/client/service/FalkonryService.java [hunk 2]
```diff
@@ -253,7 +307,64 @@ public InputStatus addInputFromStream(String id, ByteArrayInputStream stream, Ma
...
 			throws Exception {
 		ObjectMapper mapper = new ObjectMapper();
-		String url = "/assessment/" + id + "/facts";
+		String url = "/assessment/" + id + "/facts?";
+		
+		Boolean firstReqParam = true;
+        
+        
+        
+        if (options.containsKey("startTimeIdentifier")) {
```
Comment (aniket-amrutkar) [inline src/main/java/com/falkonry/client/service/FalkonryService.java]: create a separate function for generating url

File: README.md (hunk 1 above)
Comment (aniket-amrutkar) [reply README.md]: Mention single entity and multi Entity datastream clearly in the readme and test cases

**Comment by phagunbaya** [comment]
@Dilpesh What's the update on this PR?

File: src/test/java/com/falkonry/TestAddAndGetFacts.java [hunk 3]
```diff
@@ -5,327 +5,514 @@
...
-		Iterator<Datastream> itr = datastreams.iterator();
-		while (itr.hasNext()) {
-			Datastream ds = itr.next();
-			falkonry.deleteDatastream(ds.getId());
-		}
-	}
+    Falkonry falkonry = null;
+//    String host = "https://localhost:8080";
+//    String token = "auth-token";
+    String host = "https://staging.falkonry.ai";
```
Comment (aniket-amrutkar) [inline src/test/java/com/falkonry/TestAddAndGetFacts.java]: Remove this

File: src/main/java/com/falkonry/client/service/FalkonryService.java [hunk 4]
```diff
@@ -250,15 +250,83 @@ public InputStatus addInputFromStream(String id, ByteArrayInputStream stream, Ma
...
 		ObjectMapper mapper = new ObjectMapper();
-		String url = "/assessment/" + id + "/facts";
+		String url=  getAddFactsUrl(id, options);
 		byte[] data_bytes = IOUtils.toByteArray(stream);
 		String status = this.httpService.upstream(url, data_bytes);
 		return mapper.readValue(status, InputStatus.class);
 	}
+	
+	private String getAddFactsUrl(String assessmentId, Map<String, String> options)throws Exception {
+
```
Comment (aniket-amrutkar) [inline src/main/java/com/falkonry/client/service/FalkonryService.java]: Ident the code

File: src/test/java/com/falkonry/TestAddFactsStream.java [hunk 5]
```diff
@@ -14,197 +13,353 @@
...
-		Iterator<Datastream> itr = datastreams.iterator();
-		while (itr.hasNext()) {
-			Datastream ds = itr.next();
-			falkonry.deleteDatastream(ds.getId());
-		}
-	}
+    Falkonry falkonry = null;
+//    String host = "https://localhost:8080";
+//    String token = "auth-token";
+    String host = "https://staging.falkonry.ai";
```
Comment (aniket-amrutkar) [inline src/test/java/com/falkonry/TestAddFactsStream.java]: remove this

**Pull request merged**